src/char/int_rodex.c

```diff
@@ -0,0 +1,629 @@
+				|| SQL_ERROR == SQL->StmtBindColumn(stmt_items, 10, SQLDT_INT16, &it.option[0].index, 0, NULL, NULL)
+				|| SQL_ERROR == SQL->StmtBindColumn(stmt_items, 11, SQLDT_INT16, &it.option[0].value, 0, NULL, NULL)
+				|| SQL_ERROR == SQL->StmtBindColumn(stmt_items, 12, SQLDT_INT16, &it.option[1].index, 0, NULL, NULL)
+				|| SQL_ERROR == SQL->StmtBindColumn(stmt_items, 13, SQLDT_INT16, &it.option[1].value, 0, NULL, NULL)
+				|| SQL_ERROR == SQL->StmtBindColumn(stmt_items, 14, SQLDT_INT16, &it.option[2].index, 0, NULL, NULL)
+				|| SQL_ERROR == SQL->StmtBindColumn(stmt_items, 15, SQLDT_INT16, &it.option[2].value, 0, NULL, NULL)
+				|| SQL_ERROR == SQL->StmtBindColumn(stmt_items, 16, SQLDT_INT16, &it.option[3].index, 0, NULL, NULL)
+				|| SQL_ERROR == SQL->StmtBindColumn(stmt_items, 17, SQLDT_INT16, &it.option[3].value, 0, NULL, NULL)
+				|| SQL_ERROR == SQL->StmtBindColumn(stmt_items, 18, SQLDT_INT16, &it.option[4].index, 0, NULL, NULL)
+				|| SQL_ERROR == SQL->StmtBindColumn(stmt_items, 19, SQLDT_INT16, &it.option[4].value, 0, NULL, NULL)
+				|| SQL_ERROR == SQL->StmtBindColumn(stmt_items, 20, SQLDT_INT, &it.expire_time, 0, NULL, NULL)
+				|| SQL_ERROR == SQL->StmtBindColumn(stmt_items, 21, SQLDT_UINT8, &it.bound, 0, NULL, NULL)
+				|| SQL_ERROR == SQL->StmtBindColumn(stmt_items, 22, SQLDT_UINT64, &it.unique_id, 0, NULL, NULL)) {
+				SqlStmt_ShowDebug(stmt_items);
+			}
+
+			for (i = 0; i < RODEX_MAX_ITEM && SQL_SUCCESS == SQL->StmtNextRow(stmt_items); ++i) {
+				msg.items[i].item = it;
+				msg.items_count++;
+			}
+		}
+
+		if (msg.items_count <= 0) {
+			msg.type &= ~MAIL_TYPE_ITEM;
+		}
+
+		if (msg.zeny <= 0) {
+			msg.type &= ~MAIL_TYPE_ZENY;
+		}
+
+#if PACKETVER < 20160601
+		// NPC Message Type isn't supported in old clients
+		msg.type &= ~MAIL_TYPE_NPC;
+#endif
+
+		++count;
+		VECTOR_ENSURE(*mails, 1, 1);
+		VECTOR_PUSH(*mails, msg);
+		memset(&msg, 0, sizeof(struct rodex_message));
+	}
+
+	SQL->FreeResult(inter->sql_handle);
+
+	ShowInfo("rodex load complete from DB - id: %d (total: %d)\n", char_id, count);
+	return count;
+}
+
+// Loads new mails of this char_id/account_id
+static int inter_rodex_refresh_fromsql(int char_id, int account_id, int8 opentype, int64 mail_id, struct rodex_maillist *mails)
+{
+	int i, count = 0;
+	struct rodex_message msg = { 0 };
+	struct SqlStmt *stmt = SQL->StmtMalloc(inter->sql_handle);
+	struct SqlStmt *stmt_items = SQL->StmtMalloc(inter->sql_handle);
+
+	nullpo_retr(false, stmt);
+	nullpo_retr(false, stmt_items);
+
+	nullpo_retr(-1, mails);
+
+	switch (opentype) {
+	case RODEX_OPENTYPE_MAIL:
+		if (SQL_ERROR == SQL->StmtPrepare(stmt,
+			"SELECT `id`, `sender_name`, `sender_id`, `receiver_name`, `receiver_id`, `receiver_accountid`,"
+			"`title`, `body`, `zeny`, `type`, `is_readed`, `send_date`, `expire_date`, `weight`"
+			"FROM `rodex_mail` WHERE `expire_date` > '%d' AND `receiver_id` = '%d' AND `id` > '%"PRId64"'"
```

Comment:
This query uses `expire_date` (which isn't part of an index) in the WHERE condition.